src/goto-programs/remove_function_pointers.cpp

```diff
@@ -348,6 +375,41 @@ void remove_function_pointerst::remove_function_pointer(
     }
   }
 
+  // we only handle cases where the LHS is a symbol or a member of a struct
+  if(pointer.id() == ID_symbol || pointer.id() == ID_member)
+  {
+    irep_idt identifier;
+    if(pointer.id() == ID_symbol)
+      identifier =
+        to_symbol_expr(to_dereference_expr(code.function()).pointer())
+          .get_identifier();
+    else
+    {
+      identifier = get_member_identifier(to_member_expr(pointer));
+    }
+
+    if(function_pointer_map.find(identifier) != function_pointer_map.end())
+    {
+      status() << "Replacing function pointer from map of possible assigns\n";
+      if(
+        function_pointer_map[identifier].find(UNKNOWN_ASSIGN) ==
+        function_pointer_map[identifier].end())
+      {
+        for(auto func_itr = functions.begin(); func_itr != functions.end();)
+        {
+          if(
+            function_pointer_map[identifier].find(func_itr->get_identifier()) ==
```

Comment:
Can there be anything in `function_pointer_map[identifier]` but not in `functions`? If not, suggest just `functions = function_pointer_map[identifier]`, or something to that effect if their types don't quite match